src/14_legendre.jl

```diff
@@ -0,0 +1,362 @@
+
+```math
+L_n(x)
+=\frac{e^x}{n!} \frac{\mathrm{d}^n}{\mathrm{d}x^n} \left(e^{-x} x^n\right)
+\quad \text{for} \quad
+x \geq 0, \; n = 0, 1, 2, \dots
+```
+
+External links: [DLMF](https://dlmf.nist.gov/18.3.T1),
+[Wikipedia](https://en.wikipedia.org/wiki/Laguerre_polynomials).
+"""
+function laguerrel(n::Integer, x)
+    (n < 0) && throw(DomainError(n, "must be non-negative"))
+
+    ABC_recurrence(n, x,
+        ( (-one(x)/m, 2-one(x)/m, 1-one(x)/m) for m=2:n),               # (A_m, B_m, C_m)
+        one(x),                                                         # P_0(x)
+        one(x)-x)                                                       # P_1(x)
+end
+
+@doc raw"""
+    legendrep(n[, m], x)
+
+Evaluate the Legendre polynomial ``P_n(x)`` of degree ``n`` at position ``x``.
+If the order ``m`` is supplied then the associated Legendre polynomial ``P_n^{(m)}(x)`` is
+computed.
+Their definitions are given by
+```math
+\begin{align}
+    P_n^{(m)}(x)
+    &=
+    (-1)^m (1-x^2)^\frac{m}{2} \frac{\mathrm{d}^m}{\mathrm{d}x^m} \left( P_n(x) \right)
+    \,, \quad m = 0, 1, \dots, n \,, \; x \in [-1,1]
+    \\
+    P_n(x)
+    &=
+    \begin{cases}
+        1
+        & \text{for} \quad n = 0
+        \\
+        x
+        & \text{for} \quad n = 1
+        \\
+        \frac{2n-1}{n} x P_{n-1}(x) - \frac{n-1}{n} P_{n-2}(x)
+        & \text{for} \quad n \geq 2
+        \,.
+    \end{cases}
+\end{align}
+```
+The Legendre polynomial ``P_n(x)`` are also called Legendre polynomial of first kind,
+respectively ``P_n^{(m)}(x)`` associated Legendre functions of first kind.
+
+External links: [DLMF - Legendre polynomial](https://dlmf.nist.gov/14.7.E1),
+[DLMF - associated Legendre polynomial](https://dlmf.nist.gov/14.7.E8),
+[Wikipedia - Legendre polynomial](https://en.wikipedia.org/wiki/Legendre_polynomials),
+[Wikipedia - associated Legendre polynomial](https://en.wikipedia.org/wiki/Associated_Legendre_polynomials).
+"""
+function legendrep(n::Integer, x)
+    (n < 0) && throw(DomainError(n, "must be non-negative"))
+
+    ABC_recurrence(n, x,
+        ( ((2m-one(x))/m, zero(x), (m-one(x))/m) for m=2:n),            # (A_m, B_m, C_m)
+        one(x),                                                         # P_0(x)
+        x)                                                              # P_1(x)
+end
+function legendrep(n::Integer, m::Integer, x)
+
+    # special case: m=0  =>  P_n^m = P_n
+    # Legendre polynomial P_n(x) can be evaluated outside of [-1,1].
+    # Thus, we use that routine before checking x interval for associated Legendre polynomial P_n^m(x).
+    (0 == m) && return legendrep(n, x)
+
+    # general argument checks
+    (n < 0)          && throw(DomainError(n, "must be non-negative"))
+    (m < 0 || m > n) && throw(DomainError(m, "must be in the range m = 0,..,n"))
+
+    # step 1: compute P_m^m(x)
+    p = one(x)                                                      # = P_0^0(x)
```

Comment:
Not type-stable.  Should be `p = float(one(x))`.

Reply:
included in commit [f05cab9](https://github.com/JuliaMath/SpecialFunctions.jl/pull/175/commits/f05cab95f1bd88b6df5f16a94131973bfee6a12a).